File: frontend/src/types/Simulation.ts
// Matches backend: FinalProject.Models.SimulationPath
export interface SimulationPath {
  id: number;
  simulationId: number;
  nodeId: number;
  stepOrder: number;
}

// Matches backend: FinalProject.Models.Simulation
// algorithm is serialized as string due to JsonStringEnumConverter on the backend
export interface Simulation {
  id: number;
  sourceNodeId: number;
  destinationNodeId: number;
  algorithm: string;
  totalCost: number;
  executionTimeMs: number;
  visitedNodes: number;
  edgeRelaxations: number;
  pathFound: boolean;
  createdAt: string;
  pathNodes: SimulationPath[];
}

File: backend/Models/Simulation.cs
using System.ComponentModel.DataAnnotations;

namespace FinalProject.Models
{
    public class Simulation
    {
        public int Id { get; set; }

        public int SourceNodeId { get; set; }

        public int DestinationNodeId { get; set; }

        public AlgorithmType Algorithm { get; set; }

        public double TotalCost { get; set; }

        public double ExecutionTimeMs { get; set; }

        public int VisitedNodes { get; set; }

        public int EdgeRelaxations { get; set; }

        public bool PathFound { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<SimulationPath> PathNodes { get; set; } = new List<SimulationPath>();
    }
}